src/script.ts

```diff
@@ -111,37 +128,43 @@ const SUPPORTED_CODEMIRROR_SITES = [
   { pattern: /https:\/\/(.*\.)?codeshare\.io(\/.*)?/, multiplayer: true },
 ];
 
-Object.defineProperty(window, 'CodeMirror', {
-  get: function () {
-    return this._codeium_CodeMirror;
-  },
-  set: function (cm?: { version?: string }) {
-    this._codeium_CodeMirror = cm;
-    if (!cm?.version?.startsWith('5.')) {
-      console.warn("Codeium doesn't support CodeMirror 6");
-      return;
-    }
-    // We rely on the fact that the Jupyter variable is defined first.
-    if (Object.prototype.hasOwnProperty.call(this, 'Jupyter')) {
-      const jupyterState = jupyterInject(extensionId, this.Jupyter);
-      addListeners(cm as CodeMirror, jupyterState.codeMirrorManager);
-    } else {
-      let injectCodeMirror = false;
-      let multiplayer = false;
-      for (const pattern of SUPPORTED_CODEMIRROR_SITES) {
-        if (pattern.pattern.test(window.location.href)) {
+let injectCodeMirror = false;
+
+const addCodeMirrorInject = () =>
```

Comment:
Let's call this one `addCodeMirror5GlobalInject` and the other one `addCodeMirror5LocalInject`, since both only work on CodeMirror 5.